src/api-object.ts

```diff
@@ -45,7 +45,29 @@ export interface GroupVersionKind {
   readonly kind: string;
 }
 
+const API_OBJECT_SYMBOL = Symbol.for('cdk8s.ApiObject');
+
 export class ApiObject extends Construct {
+
+  /**
+   * Return whether the given object is an `ApiObject`.
+   *
+   * We do attribute detection since we can't reliably use 'instanceof'.
+
+   * @param o The object to check
+   */
+  static isApiObject(o: unknown): o is ApiObject {
+    return o != null && typeof o === 'object' && API_OBJECT_SYMBOL in o;
+  }
+
+  /**
+   * Implements `instanceof ApiObject` using the more reliable `ApiObject.isApiObject` static method
+   *
+   * @param o The object to check
```

Comment:
Build log suggests you should:

```suggestion
   * @param o The object to check
   * @internal
```

To make jsii happy.

Reply:
Thanks, applied here as well as on `ApiObject`